### ISSUE TITLE
Free space calculation on Apple File System (APFS)

### ISSUE THREAD

**Issue by mvasin** [djberg96/sys-filesystem#28]
[This thread on SO](https://stackoverflow.com/questions/4508692/get-available-diskspace-in-ruby) suggests a bunch of ways to measure free disk space, like

```
require 'sys/filesystem'

stat = Sys::Filesystem.stat("/")
mb_available = stat.block_size * stat.blocks_available / 1024 / 1024
```
or
```
mb_available = stat.bytes_free / 1024 / 1024 / 1024
```

It does work on linux and used to work on a mac, but on OS X 10.13 with APFS the result is the occupied space, not the free space.

**Comment by djberg96** [comment]
@mvasin I haven't installed 10.13 yet, so I'm not sure what underlying changes they've made. Without a PR, I'm afraid this will have to wait for now, but thanks for the report.

**Comment by djberg96** [comment]
The `bytes_free` and `bytes_used` seems to be off by a factor of 4, but the the `percent_used` method still works, since they're both off by the same factor.

I'll have to dig into their docs, assuming any exist, to see if I can confirm this.

**Comment by djberg96** [comment]
I think the key is to use `fragment_size` on Mac (any filesystem) instead of `block_size`. On my HFS system they're both 4096, while on APFS the fragment size is 4096 but the block size is 4x that. A few scattered posts on StackOverflow and elsewhere seem to confirm this.

**Comment by djberg96** [comment]
@mvasin Fixed in 1.1.9. Thanks for the report!
